core/src/main/java/org/apache/iceberg/encryption/Ciphers.java

```diff
@@ -101,37 +117,52 @@ public AesGcmDecryptor(byte[] keyBytes) {
     }
 
     public byte[] decrypt(byte[] ciphertext, byte[] aad) {
-      int plainTextLength = ciphertext.length - GCM_TAG_LENGTH - NONCE_LENGTH;
+      return decrypt(ciphertext, 0, ciphertext.length, aad);
+    }
+
+    public byte[] decrypt(
```

Comment:
I think that `TestCiphers` needs to have tests validating failure modes:
* If the supplied AAD is incorrect, decryption fails
* If the ciphertext is modified, decryption fails

Right now, only the happy path is tested, which doesn't validate the intended authentication behavior using the tag.